website/docs/d/karbon_cluster.html.markdown

```diff
@@ -0,0 +1,102 @@
+page_title: "NUTANIX: nutanix_karbon_cluster"
+sidebar_current: "docs-nutanix-datasource-karbon-cluster"
+description: |-
+ Describes a Karbon Cluster
+---
+
+# nutanix_karbon_cluster
+
+Describes Karbon Clusters
+
+## Example Usage
+
+```hcl
+data "nutanix_karbon_cluster" "cluster" {
+   cluster_id = "<YOUR-CLUSTER-ID>"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `karbon_cluster_id`: Represents karbon cluster uuid
+* `karbon_cluster_name`: Represents the name of karbon cluster
+
+## Attribute Reference
+
+The following arguments are supported:
+
+* `name`: - The name for the k8s cluster.
+* `wait_timeout_minutes`
+* `version`: - K8s version of the cluster.
+* `storage_class_config`
+* `single_master_config`: - Configuration of a single master node.
+* `active_passive_config`: - The active passive mode uses the Virtual Router Redundancy Protocol (VRRP) protocol to provide high availability of the master.
+* `external_lb_config`: - The external load balancer configuration in the case of a multi-master-external-load-balancer type master deployment.
+* `private_registry`: - .
+* `etcd_node_pool`: - Configuration of the node pools that the nodes in the etcd cluster belong to. The etcd nodes require a minimum of 8,192 MiB memory and 409,60 MiB disk space.
+* `master_node_pool`: - .
+* `cni_config`: - K8s cluster networking configuration. The flannel or the calico configuration needs to be provided.
+
+### Storage Class Config
+
+The storage_class_config attribute supports the following:
+
+* `name`: - The name of the storage class.
+* `reclaim_policy` - Reclaim policy for persistent volumes provisioned using the specified storage class.
+* `volumes_config.#.file_system` - Karbon uses either the ext4 or xfs file-system on the volume disk.
+* `volumes_config.#.flash_mode`
+* `volumes_config.#.password` - TPassword of the Prism Element user that the API calls use to provision volumes.
+* `volumes_config.#.prism_element_cluster_uuid` - The universally unique identifier (UUID) of the Prism Element cluster.
+* `volumes_config.#.storage_container` - TName of the storage container the storage container uses to provision volumes.
+* `volumes_config.#.username` - TThe username of the Prism Element user that the API calls use to provision volumes.
+
+
+### External LB Config
+
+The external load balancer configuration in the case of a multi-master-external-load-balancer type master deployment.
+
+* `external_lb_config.#.external_ipv4_address`: The external load balancer IPV4 address.
+* `external_lb_config.#.master_nodes_config`: The configuration of the master nodes.
+* `external_lb_config.#.master_nodes_config.ipv4_address`: The IPV4 address to assign to the master.
+* `external_lb_config.#.master_nodes_config.node_pool_name`: The name of the node pool in which this master IPV4 address will be used.
+
+### private_registry
+User inputs of storage configuration parameters for VMs.
+
+* `private_registry`: - .
```

Comment:
List of private registries

Note: Should we rename the parameter to make it clear that it is a list?

Reply:
@marinsalinas should we?